doc/kube-api-test.md

```diff
@@ -0,0 +1,153 @@
+Linux, Windows, Mac is supported.
+
+## Usage
+
+Include dependency:
+
+```xml
+<dependency>
+    <groupId>io.fabric8</groupId>
+    <artifactId>kube-api-test</artifactId>
+    <version>[version]</version>
+    <scope>test</scope>
+</dependency>
+```
+
+### In Unit Tests
+
+See sample unit
+test [here](https://github.com/csviri/kubernetes-client/blob/main/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/JUnitExtensionSimpleCaseTest.java)
+
+```java
+
+@EnableKubeAPIServer
+class JUnitExtensionSimpleCaseTest {
+
+    // Use @KubeConfig annotation to inject kube config yaml to init any client
+    @KubeConfig
+    static String kubeConfigYaml;
+
+    @Test
+    void simpleTestWithTargetVersion() {
+        var client = new KubernetesClientBuilder()
+                .withConfig(Config.fromKubeconfig(kubeConfigYaml))
+                .build();
+
+        client.resource(TestUtils.testConfigMap()).create();
+        var cm = client.resource(TestUtils.testConfigMap()).get();
+
+        Assertions.assertThat(cm).isNotNull();
+    }
+}
+```
+
+### Public API
+
+The underlying API can be used directly.
+See [KubeApiServer](https://github.com/csviri/kubernetes-client/blob/main/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java#L28-L28)
+
+See
+it's [usage in a test](https://github.com/csviri/kubernetes-client/blob/main/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubeApiServerTest.java#L31-L31).
+
+```java
+class KubeApiServerTest {
+    
+    @Test
+    void apiServerTest() {
+        var kubeApi = new KubeAPIServer();
+        kubeApi.start();
+
+        var client =  new KubernetesClientBuilder()
+                .withConfig(Config.fromKubeconfig(kubeApi.getKubeConfigYaml()))
+                .build();
+        
+        client.resource(TestUtils.testConfigMap()).create();
+        
+        var cm = client.resource(TestUtils.testConfigMap()).get();
+        Assertions.assertThat(cm).isNotNull();
+        
+        kubeApi.stop();
+    }
+}
+```
+
+### Fabric8 Client Injection Support
+
+Use dependency:
+
+```xml
+<dependency>
+    <groupId>io.fabric8</groupId>
+    <artifactId>kube-api-test-client-inject</artifactId>
+    <version>[version]</version>
+    <scope>test</scope>
+</dependency>
+```
+
+The client can be directly injected to the test. See sample test [here](https://github.com/csviri/kubernetes-client/blob/main/junit/kube-api-test/client-inject/src/test/java/io/fabric8/kubeapitest/junit/sample/JUnitFabric8ClientInjectionTest.java#L28-L28).
+
+```java
+
+@EnableKubeAPIServer
+class JUnitFabric8ClientInjectionTest {
+
+    static KubernetesClient client;
+   
+    // emitted code     
+}  
+```
+
+### Configuration Options
+
+See available configuration options documented in [KubeAPIServerConfig](https://github.com/csviri/kubernetes-client/blob/main/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServerConfig.java)
```

Comment:
It would be nice to have these available as a table (just like we have in [Configuring the Client](https://github.com/fabric8io/kubernetes-client#configuring-the-client) section)

Reply:
I have mixed feelings about that, would help for sure, to my taste this is readable / usable enough but would worth it yes.

Reply:
Umm, okay. It's totally up to you.